### PR TITLE
Resolve deprecations (+ some fixes)

### LINKED ISSUE #168: Nextcloud 19 errors
### Steps to reproduce
1. Update to nextcloud 19.0.0.12
2. Install OCdownloader 1.7.8
3. Load any page of this app

### Expected behaviour
My logs should be "empty"
**Maybe** related to my issue https://github.com/e-alfred/ocdownloader/issues/165#issuecomment-641436706
### Actual behaviour
My logs are flooded with errors from this app :
```
{"reqId":"3vixaGmWqD3mHZCDehjj","level":3,"time":"2020-06-09T16:20:42+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"GET","url":"/index.php/apps/ocdownloader/api/queue/get?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLBinary as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#64","userAgent":"Mozilla/5.0","version":"19.0.0.12"}
{"reqId":"3vixaGmWqD3mHZCDehjj","level":3,"time":"2020-06-09T16:20:42+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"GET","url":"/index.php/apps/ocdownloader/api/queue/get?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLAudioFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#69","userAgent":"Mozilla/5.0","version":"19.0.0.12"}
{"reqId":"3vixaGmWqD3mHZCDehjj","level":3,"time":"2020-06-09T16:20:42+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"GET","url":"/index.php/apps/ocdownloader/api/queue/get?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLVideoFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#74","userAgent":"Mozilla/5.0","version":"19.0.0.12"}
{"reqId":"KtkN6MBcTDfx98Krv7ZH","level":3,"time":"2020-06-09T16:22:33+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLBinary as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#64","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
{"reqId":"KtkN6MBcTDfx98Krv7ZH","level":3,"time":"2020-06-09T16:22:33+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLAudioFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#69","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
{"reqId":"KtkN6MBcTDfx98Krv7ZH","level":3,"time":"2020-06-09T16:22:33+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLVideoFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#74","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
{"reqId":"Z0nYKXiXeFiNZHJxJhrz","level":3,"time":"2020-06-09T16:35:07+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLBinary as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#64","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
{"reqId":"Z0nYKXiXeFiNZHJxJhrz","level":3,"time":"2020-06-09T16:35:07+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLAudioFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#69","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
{"reqId":"Z0nYKXiXeFiNZHJxJhrz","level":3,"time":"2020-06-09T16:35:07+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLAudioFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#71","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
{"reqId":"Z0nYKXiXeFiNZHJxJhrz","level":3,"time":"2020-06-09T16:35:07+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLVideoFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#74","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
{"reqId":"Z0nYKXiXeFiNZHJxJhrz","level":3,"time":"2020-06-09T16:35:07+00:00","remoteAddr":"***HIDDEN IP***","user":"***HIDDEN USER***","app":"PHP","method":"POST","url":"/index.php/apps/ocdownloader/api/add?format=json","message":"Accessing static property OCA\\ocDownloader\\Controller\\Lib\\API::$YTDLVideoFormat as non static at /var/www/cloud/apps/ocdownloader/controller/lib/api.php#76","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.142 Safari/537.36","version":"19.0.0.12"}
```

### Server configuration

**Operating system**:
Ubuntu 18.04.4 LTS
**Web server:**
nginx/1.19.0 mainline ( from nginx.org APT repository )
**Database:**
mariadb  Ver 15.1 Distrib 10.1.44-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2
**PHP version:**
PHP 7.4.6 (cli) (built: May 14 2020 10:02:44) ( NTS ) ( ondrej ppa APT repository )
**Nextcloud version:** (see Nextcloud admin page)
19.0.0.12
**Updated from an older Nextcloud/ownCloud or fresh install:**
18.0.5
**Where did you install Nextcloud from:**
Zipped archive from nextcloud.com
**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.5.0
  - activity: 2.12.0
  - admin_audit: 1.9.0
  - apporder: 0.10.0
  - bruteforcesettings: 1.6.0
  - calendar: 2.0.3
  - checksum: 0.4.4
  - cloud_federation_api: 1.2.0
  - comments: 1.9.0
  - contacts: 3.3.0
  - contactsinteraction: 1.0.0
  - cospend: 0.5.5
  - dav: 1.15.0
  - deck: 1.0.2
  - event_update_notification: 1.0.2
  - extract: 1.2.4
  - federatedfilesharing: 1.9.0
  - federation: 1.9.0
  - files: 1.14.0
  - files_accesscontrol: 1.9.0
  - files_automatedtagging: 1.9.0
  - files_downloadactivity: 1.8.0
  - files_external: 1.10.0
  - files_linkeditor: 1.1.0
  - files_markdown: 2.3.0
  - files_mindmap: 0.0.22
  - files_pdfviewer: 1.8.0
  - files_rightclick: 0.16.0
  - files_sharing: 1.11.0
  - files_texteditor: 2.11.0
  - files_trackdownloads: 1.8.0
  - files_trashbin: 1.9.0
  - files_versions: 1.12.0
  - files_videoplayer: 1.8.0
  - firstrunwizard: 2.8.0
  - forms: 1.1.1
  - keeweb: 0.6.2
  - logreader: 2.4.0
  - lookup_server_connector: 1.7.0
  - metadata: 0.11.1
  - music: 0.14.1
  - news: 14.1.11
  - nextcloud_announcements: 1.8.0
  - notes: 3.5.0
  - notifications: 2.7.0
  - oauth2: 1.7.0
  - ocdownloader: 1.7.8
  - ocsms: 2.1.7
  - password_policy: 1.9.1
  - phonetrack: 0.6.3
  - photos: 1.1.0
  - polls: 1.4.3
  - privacy: 1.3.0
  - provisioning_api: 1.9.0
  - quota_warning: 1.8.0
  - ransomware_protection: 1.7.0
  - richdocumentscode: 4.2.403
  - serverinfo: 1.9.0
  - settings: 1.1.0
  - sharebymail: 1.9.0
  - sharelisting: 0.3.0
  - sharingpath: 0.2.2
  - socialsharing_email: 2.0.0
  - socialsharing_twitter: 2.0.0
  - spreed: 9.0.1
  - support: 1.2.1
  - survey_client: 1.7.0
  - suspicious_login: 3.1.0
  - systemtags: 1.9.0
  - talk_simple_poll: 1.1.1
  - tasks: 0.13.1
  - telephoneprovider: 1.0.3
  - theming: 1.10.0
  - twofactor_admin: 2.1.0
  - twofactor_backupcodes: 1.8.0
  - twofactor_email: 1.0.1
  - twofactor_gateway: 0.16.0
  - twofactor_nextcloud_notification: 2.3.0
  - twofactor_totp: 4.1.3
  - twofactor_u2f: 5.1.0
  - updatenotification: 1.9.0
  - video_converter: 0.1.4
  - viewer: 1.3.0
  - workflow_pdf_converter: 1.4.0
  - workflow_script: 1.4.0
  - workflowengine: 2.1.0
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "auth.bruteforce.protection.enabled": true,
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "filelocking.enabled": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 1.5,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "0": {
            "verify_peer_off": false
        },
        "version": "19.0.0.12",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "onlyoffice": "",
        "installed": true,
        "enable_previews": true,
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\Movie",
            "OC\\Preview\\PDF",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown"
        ],
        "app_install_overwrite": [
            "limit_login_to_ip",
            "onlyoffice",
            "shorten",
            "keeweb",
            "forms",
            "ocsms",
            "twofactor_email",
            "checksum",
            "contacts",
            "sharerenamer",
            "socialsharing_email",
            "uploaddetails",
            "socialsharing_twitter",
            "sharingpath",
            "jsloader",
            "external",
            "telephoneprovider",
            "browser_warning",
            "beame_insta_ssl"
        ],
        "maintenance": false,
        "twofactor_enforced": "true",
        "has_internet_connection": "true",
        "twofactor_enforced_groups": [],
        "twofactor_enforced_excluded_groups": [],
        "updater.release.channel": "stable",
        "theme": "",
        "loglevel": 2,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mysql.utf8mb4": true,
        "trashbin_retention_obligation": "auto",
        "versions_retention_obligation": "auto"
    }
}
```
</details>

**Are you using external storage, if yes which one (currently not supported by ocDownloader):** local/smb/sftp/...
None
**Are you using encryption (currently not supported by ocDownloader):** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
None

#### ocDownloader configuration:

**Which downloader are you using:** Curl/Aria2
Aria2
**Do you use a proxy:** Yes/No
No
**Which protocols do you allow**: FTP/HTTP/Bittorrent/Youtube
All protocols
#### Aria2c configuration (remove if not used):
Default ( followed README.md instructions )

### PR DESCRIPTION
**Pull request by weeman1337**
To make ocdownloader a bit more future proof I have worked on resolving deprecations (beside fixing some issues):

- Added an [editorconfig](https://editorconfig.org/) with some basic settings.  
Hopefully makes future contributions a bit easier regarding very basic code style (e.g.  indentation).
- Fixed static/instance variable usages, missing imports
- Improves magic method editor support by introducing doc blocks
- Resolved NC 19 deprecations
- Replaced JS `eval()` to be able to migrate away from a deprecated content security policy
- Migrated personal and amin settings to the [new API](https://docs.nextcloud.com/server/19/developer_manual/app/settings.html)
- Catch `Throwable` instead of `Exception` because that catches errors as well

This PR doesn't change any functionality. It contains only maintenance.

Fixes #168

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# https://editorconfig.org
```
Comment (weeman1337) [inline .editorconfig:1]: @e-alfred EditorConfig is supported by many editors and IDEs.

Would be nice to add this. Please check whether the settings are fine for you.

File: controller/lib/aria2.php [hunk 2]
```diff
@@ -10,14 +10,23 @@
  */
 namespace OCA\ocDownloader\Controller\Lib;
 
-use \OCA\ocDownloader\Controller\Lib\Settings;
-
+/**
+ * @method static addTorrent($hash, $p1, $params)
```
Comment (weeman1337) [inline controller/lib/aria2.php:14]: These doc lines improve editor support for the magic methods.

File: .editorconfig (hunk 1 above)
Comment (e-alfred) [reply .editorconfig:1]: I think an EditorConfig would surely help improve working on this app.

**Comment by e-alfred** [comment]
Thanks for your help, I will merge this and make a new release soon. If you are interested, I could make you a collaborator for this repository.

**Comment by weeman1337** [comment]
> If you are interested, I could make you a collaborator for this repository.

Hi @e-alfred I would be happy to help! :+1:

**Pull request merged**